components/Events/Filters.js

```diff
@@ -0,0 +1,236 @@
+import { faChevronUp, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { useState, useReducer, React } from 'react';
+import events from '../../data/event';
+import styles from '../../styles/components/Events/Filters.module.scss';
+
+export default function filters(props) {
+  function reducer(state, action) {
+    switch (action.type) {
+      case 'comms':
+        return { ...state, comms: { ...state.comms, [action.payload.key]: action.payload.val } };
+      case 'days':
+        return { ...state, days: { ...state.days, [action.payload.key]: action.payload.val } };
+      case 'times':
+        return { ...state, times: { ...state.times, [action.payload.key]: action.payload.val } };
+      case 'loc':
+        return { ...state, loc: { ...state.loc, [action.payload.key]: action.payload.val } };
+      default:
+        throw new Error();
+    }
+  }
+  const initialState = {
+    comms: {
+      'studio': false,
+      'icpc': false,
+      'design': false,
+      'cyber': false,
+      'teach-la': false,
+      'w': false,
+      'ai': false,
+      'hack': false,
+      'board': false,
+    },
+    days: {
+      Sunday: false,
+      Monday: false,
+      Tuesday: false,
+      Wednesday: false,
+      Thursday: false,
+      Friday: false,
+      Saturday: false,
+    },
+    times: {
+      from: '',
+      to: '',
+    },
+    loc: {
+      'Online': false,
+      'In-Person': false,
+    },
+  };
+  const [filterVals, dispatch] = useReducer(reducer, initialState);
+  const [isFilterOpen, setIsFilterOpen] = useState(false);
+  const [searchValue, setSearchValue] = useState('');
+
+  const dateToMinutes = (date) => {
+    const itemMins =
+      date.getHours() * 60 + date.getMinutes();
+    return itemMins;
+  };
+  const filterTime = (filterEvents) => {
```

Comment:
`filterEvents` is a bit confusing, it sounds like filter is an event. I think `eventsToFilter` also works, but it's a bit long
```suggestion
  const filterTime = (events) => {
```